src/main/java/org/opentripplanner/gtfs/mapping/LocationMapper.java

```diff
@@ -5,53 +5,70 @@
 import java.util.Map;
-import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.operation.valid.IsValidOp;
+import org.onebusaway.gtfs.model.Location;
 import org.opentripplanner.framework.collection.MapUtils;
 import org.opentripplanner.framework.geometry.GeometryUtils;
 import org.opentripplanner.framework.geometry.UnsupportedGeometryException;
 import org.opentripplanner.framework.i18n.NonLocalizedString;
+import org.opentripplanner.graph_builder.issue.api.DataImportIssueStore;
+import org.opentripplanner.graph_builder.issue.api.Issue;
 import org.opentripplanner.transit.model.site.AreaStop;
 import org.opentripplanner.transit.service.StopModelBuilder;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /** Responsible for mapping GTFS Location into the OTP model. */
 public class LocationMapper {
 
-  private static final Logger LOG = LoggerFactory.getLogger(LocationMapper.class);
-
-  private final Map<org.onebusaway.gtfs.model.Location, AreaStop> mappedLocations = new HashMap<>();
+  private final Map<Location, AreaStop> mappedLocations = new HashMap<>();
   private final StopModelBuilder stopModelBuilder;
+  private final DataImportIssueStore issueStore;
 
-  public LocationMapper(StopModelBuilder stopModelBuilder) {
+  public LocationMapper(StopModelBuilder stopModelBuilder, DataImportIssueStore issueStore) {
     this.stopModelBuilder = stopModelBuilder;
+    this.issueStore = issueStore;
   }
 
-  Collection<AreaStop> map(Collection<org.onebusaway.gtfs.model.Location> allLocations) {
+  Collection<AreaStop> map(Collection<Location> allLocations) {
     return MapUtils.mapToList(allLocations, this::map);
   }
 
   /** Map from GTFS to OTP model, {@code null} safe. */
-  AreaStop map(org.onebusaway.gtfs.model.Location orginal) {
+  AreaStop map(Location orginal) {
     return orginal == null ? null : mappedLocations.computeIfAbsent(orginal, this::doMap);
   }
 
-  private AreaStop doMap(org.onebusaway.gtfs.model.Location gtfsLocation) {
+  private AreaStop doMap(Location gtfsLocation) {
     var name = NonLocalizedString.ofNullable(gtfsLocation.getName());
-    Geometry geometry = null;
+    var id = mapAgencyAndId(gtfsLocation.getId());
     try {
-      geometry = GeometryUtils.convertGeoJsonToJtsGeometry(gtfsLocation.getGeometry());
-    } catch (UnsupportedGeometryException e) {
-      LOG.error("Unsupported geometry type for {}", gtfsLocation.getId());
+      var geometry = GeometryUtils.convertGeoJsonToJtsGeometry(gtfsLocation.getGeometry());
+      var isValidOp = new IsValidOp(geometry);
+      if (!isValidOp.isValid()) {
+        var error = isValidOp.getValidationError();
+        issueStore.add(
+          Issue.issue(
+            "InvalidFlexAreaGeometry",
+            "GTFS flex location %s has an invalid geometry: %s at (lat: %s, lon: %s)",
+            id,
+            error.getMessage(),
+            error.getCoordinate().y,
+            error.getCoordinate().x
+          )
+        );
+      }
```

Comment:
So are there some geometries which don't throw an exception but are still considered invalid by the validator? Do these cause issues during runtime?

Reply:
The routing sometimes (mostly?) works anyway but the vector tiles library refuses to serialise these geometries.